### PR TITLE
Disable ssh-agent if it is disabled in user's SSH configuration

### PR DESCRIPTION
**Pull request by olavfosse**

### CODE REVIEW

**Comment by jacobobryant** [comment]
Can you give me some details on the motivation for this change? Do you have these options set in your own config? I read the SSH_CONFIG docs you pasted, but I'm not quite sure if the idea is that the user setting these options means they don't want to use ssh-agent _or_ if when these options are set, the ssh-agent won't work anyway so we might as well not try. If it's the former, I'm not sure that actually matters? We're not starting a long-running ssh-agent; once the bb task finishes, the agent gets killed. There is already the `:biff.tasks/skip-ssh-agent` option if someone really doesn't want the agent to be used at all for some reason (mainly I put it there in case there were bugs, so there'd be a workaround instead of having bb tasks totally broken).

**Comment by olavfosse** [comment]
On further thought I've decided this is not a good change to make so i'm closing this pr.

tl;dr:

- running bb tasks was printing an ugly ssh-agent warning
- i had ssh-agent disabled in my ssh config (probably the cause of the warning)
- setting :biff.tasks/skip-ssh-agent fixed it, but it felt wrong having a personal setting in the :dev config
- might as well infer ssh settings from user's ssh config
- on further thought, :biff.tasks/skip-ssh-agent is fine for a solo project and if there's several devs I could add a :personal or :<name> config map to config.edn.
